src/passes/Outlining.cpp

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2023 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This pass is a placeholder, encapsulating a suffix tree data structure that
+// will eventually be moved to src/support
+
+#include "ir/module-utils.h"
+#include "ir/names.h"
+#include "pass.h"
+#include "src/ir/utils.h"
+#include "stringify-walker.h"
+#include "wasm-builder.h"
+#include "wasm-stack.h"
+#include <wasm.h>
+
+namespace wasm {
+
+void StringifyWalker::walkModule(Module* module) {
+  this->pushTask(StringifyWalker::handler, nullptr);
+  ModuleUtils::iterDefinedFunctions(*module, [&](Function* func) {
+    this->emitFunctionBegin(this);
+    this->walk(func->body);
+  });
+}
+
+void StringifyWalker::scanChildren(StringifyWalker* stringify,
+                                   Expression** currp) {
+  Expression* curr = *currp;
+  switch (curr->_id) {
+    case Expression::Id::BlockId: {
```

Comment:
This looks the same as what `PostWalker<...>::scan` would do for a block. Maybe we can just call out to that scan method here for everything except `If`.